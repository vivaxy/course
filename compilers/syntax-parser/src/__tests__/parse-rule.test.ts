/**
 * @since 2020-03-05 10:55
 * @author vivaxy
 */
import parseRule from '../parse-rule';

test('parse simple rule', function() {
  expect(
    parseRule(`
String ::= /[a-z][A-Z][0-9]/;
Null ::= 'null';
NullAndUndefined ::= Null 'undefined';
NullOrUndefined ::= Null | 'undefined'
`),
  ).toMatchSnapshot();
});
