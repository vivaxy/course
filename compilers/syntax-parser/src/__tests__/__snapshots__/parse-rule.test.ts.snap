// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse simple rule 1`] = `
Rule {
  "paths": Array [
    Array [
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [],
            "type": "RegExp",
            "value": /\\^\\(\\[a-z\\]\\[A-Z\\]\\[0-9\\]\\)/,
          },
        ],
        "type": "Rule",
        "value": "String",
      },
    ],
    Array [
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [],
            "type": "String",
            "value": "null",
          },
        ],
        "type": "Rule",
        "value": "Null",
      },
    ],
    Array [
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [
              Node {
                "nextNodes": Array [],
                "type": "String",
                "value": "undefined",
              },
            ],
            "type": "Rule",
            "value": "Null",
          },
        ],
        "type": "Rule",
        "value": "NullAndUndefined",
      },
    ],
    Array [
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [],
            "type": "Rule",
            "value": "Null",
          },
          Node {
            "nextNodes": Array [],
            "type": "String",
            "value": "undefined",
          },
        ],
        "type": "Rule",
        "value": "NullOrUndefined",
      },
    ],
  ],
  "startNode": Node {
    "nextNodes": Array [
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [],
            "type": "RegExp",
            "value": /\\^\\(\\[a-z\\]\\[A-Z\\]\\[0-9\\]\\)/,
          },
        ],
        "type": "Rule",
        "value": "String",
      },
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [],
            "type": "String",
            "value": "null",
          },
        ],
        "type": "Rule",
        "value": "Null",
      },
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [
              Node {
                "nextNodes": Array [],
                "type": "String",
                "value": "undefined",
              },
            ],
            "type": "Rule",
            "value": "Null",
          },
        ],
        "type": "Rule",
        "value": "NullAndUndefined",
      },
      Node {
        "nextNodes": Array [
          Node {
            "nextNodes": Array [],
            "type": "Rule",
            "value": "Null",
          },
          Node {
            "nextNodes": Array [],
            "type": "String",
            "value": "undefined",
          },
        ],
        "type": "Rule",
        "value": "NullOrUndefined",
      },
    ],
    "type": "Start",
    "value": null,
  },
}
`;
